### PR TITLE
Add headless updater functionality.

### PR DESCRIPTION
**Pull request by elipup**
This allows selecting a specific database to upgrade, and it will update it to the latest version.
Addresses #8 and #38

### CODE REVIEW

File: database.sh [hunk 1]
```diff
@@ -752,70 +752,98 @@ check_database_info() {
...
+    if [ ! -e /tmp/.X99-lock ]; then
+        log_exec start-stop-daemon --start -b -x /usr/bin/Xvfb :99
+        export DISPLAY=:99
     fi
-    log_arg $DATABASE $VERSION
+
+    # check for plv8
+    PLV8=`psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c "select exists (select 1 from pg_extension where extname = 'plv8');"`
+    if [ "$PLV8" == "f" ]; then
+        log_exec psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c "create EXTENSION plv8;"
```
Comment (davidbeauchamp) [inline database.sh]: CREATE EXTENSION IF NOT EXISTS instead of the conditional above?

File: database.sh [hunk 2]
```diff
@@ -752,70 +752,98 @@ check_database_info() {
...
-    DATABASE=demo481
-    APP=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c \"SELECT fetchmetrictext('Application') AS application;\""`
-    log "Detected application $APP"
-    VER=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c \"SELECT fetchmetrictext('ServerVersion') AS application;\""`
-    log "Detected server version $VER"
-    UPS=`curl -s http://api.xtuple.org/upgradepath.php\?package=$APP\&fromver=3.7.0\&tover=4.8.1`
-    log "Detected upgrades $UPS"
+    check_database_info
+
+    if [ -z "$UPDATEREXEC" ]; then
```
Comment (davidbeauchamp) [inline database.sh]: are UPDATEREXEC and UPDATEPKGS expected to be environment variables, with PKGS being an array?

File: database.sh [hunk 3]
```diff
@@ -752,70 +752,95 @@ check_database_info() {
...
+        UPDATEREXEC=$(whiptail --backtitle "$( window_title )" --inputbox "Auto updater executable location" 8 60 ${HOME}/updater/utilities/AutoUpdate/xtuple_autoupdater 3>&1 1>&2 2>&3)
+        RET=$?
+        if [ $RET -ne 0 ]; then
+            return $RET
+        else
```
Comment (gilmoskowitz) [inline database.sh]: `else` is unnecessary

File: database.sh [hunk 4]
```diff
@@ -752,70 +752,95 @@ check_database_info() {
...
+            export UPDATEREXEC
+        fi
+    fi
+
+    if [ -z "$UPDATEPKGS" ]; then
+        UPDATEPKGS=$(whiptail --backtitle "$( window_title )" --inputbox "Updater packages directory" 8 60 ${HOME}/Updater/pkgs 3>&1 1>&2 2>&3)
+        RET=$?
+        if [ $RET -ne 0 ]; then
+            return $RET
+        else
```
Comment (gilmoskowitz) [inline database.sh]: `else` is unnecessary

File: database.sh [hunk 5]
```diff
@@ -752,70 +752,95 @@ check_database_info() {
...
     if [ -z "$1" ]; then
         DATABASES=()
+	   VERSIONS=()
+	   APPLICATIONS=()
 
         while read -r line; do
-            DATABASES+=("$line" "$line")
+            DATABASES+=("$line")
+		  VER=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $line -c \"SELECT fetchmetrictext('ServerVersion') AS application;\""`
+		  APP=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $line -c \"SELECT fetchmetrictext('Application') AS application;\""`
```
Comment (gilmoskowitz) [inline database.sh]: `sudo` should be unnecessary as long as `${PGUSER}` is valid.

File: database.sh (hunk 2 above)
Comment (elipup) [inline database.sh]: Yes, that's right.

**Comment by gilmoskowitz** [comment]
+1

File: database.sh [hunk 6]
```diff
@@ -752,70 +752,93 @@ check_database_info() {
...
 upgrade_database() {
 
-    msgbox "This functionality is forthcoming..."
-    return 0 # for now
-    DATABASE=demo481
-    APP=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c \"SELECT fetchmetrictext('Application') AS application;\""`
-    log "Detected application $APP"
-    VER=`sudo su - postgres -c "psql -At -U ${PGUSER} -p ${PGPORT} $DATABASE -c \"SELECT fetchmetrictext('ServerVersion') AS application;\""`
-    log "Detected server version $VER"
-    UPS=`curl -s http://api.xtuple.org/upgradepath.php\?package=$APP\&fromver=3.7.0\&tover=4.8.1`
```
Comment (Rincoln) [inline database.sh]: Is this url not currently active?  I need to determine upgrade path.

**Pull request merged**